host/src/FunctionsCustomHost/Program.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.VisualBasic;
+using Newtonsoft.Json.Linq;
+using System.IO;
+
+namespace FunctionsCustomHost
+{
+    internal class Program
+    {
+        static async Task Main(string[] args)
+        {
+            try
+            {
+                Logger.Log("Starting FunctionsCustomHost");
+
+                using var appLoader = new AppLoader();
+
+                var localSettingsJObject = await GetLocalSettingsJsonAsJObjectAsync();
+                var workerRuntime = localSettingsJObject?["Values"]?[EnvironmentVariables.FunctionsWorkerRuntime]?.Value<string>();
+                if (string.IsNullOrEmpty(workerRuntime))
+                {
+                    Logger.Log($"Environment variable '{EnvironmentVariables.FunctionsWorkerRuntime}' is not set.");
+                    return;
+                }
+                if (workerRuntime == DotnetConstants.DotnetWorkerRuntime)
+                {
+                    var isInProc8 = localSettingsJObject?["Values"]?[EnvironmentVariables.FunctionsInProcNet8Enabled]?.Value<string>();
+                    // Load host assembly for .NET 8 in proc host
+                    if (!string.IsNullOrEmpty(isInProc8) && string.Equals("1", isInProc8))
+                    {
+                        Logger.Log("Loading inproc8 host");
+                        LoadHostAssembly(appLoader, isNet8InProc: true);
+                    }
+                    else
+                    {
+                        // Load host assembly for .NET 6 in proc host
+                        Logger.Log("Loading inproc6 host");
+                        LoadHostAssembly(appLoader, isNet8InProc: false);
+                    }
+
+                }
+            }
+            catch (Exception exception)
+            {
+                Logger.Log($"An error occurred while running FunctionsCustomHost.{exception}");
+            }
+        }
+
+        private static void LoadHostAssembly(AppLoader appLoader, bool isNet8InProc)
+        {
+            var currentDirectory = AppContext.BaseDirectory;
```

Comment:
I see we use `Environment.CurrentDirectory` in another place. Let's stick with one and be consistent with that.

Reply:
I think we need AppContext.BaseDirectory though since the FunctionsCustomHost.exe would be in the root (and not the current directory for where the function app is running)